File: app/src/test/java/me/rei_m/hyakuninisshu/domain/model/karuta/KarutaIdsTest.java
/*
 * Copyright (c) 2017. Rei Matsushita
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is
 * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
 * the License for the specific language governing permissions and limitations under the License.
 */

package me.rei_m.hyakuninisshu.domain.model.karuta;

import org.junit.Before;
import org.junit.Test;

import java.util.ArrayList;
import java.util.List;

import static org.hamcrest.Matchers.is;
import static org.junit.Assert.assertThat;

public class KarutaIdsTest {

    private KarutaIds karutaIds;

    @Before
    public void setUp() throws Exception {
        karutaIds = new KarutaIds(new ArrayList<KarutaIdentifier>() {
            {
                add(new KarutaIdentifier(1));
                add(new KarutaIdentifier(2));
                add(new KarutaIdentifier(3));
            }
        });
    }

    @Test
    public void asList() throws Exception {
        List<KarutaIdentifier> expected = new ArrayList<KarutaIdentifier>() {
            {
                add(new KarutaIdentifier(1));
                add(new KarutaIdentifier(2));
                add(new KarutaIdentifier(3));
            }
        };
        assertThat(karutaIds.asList(), is(expected));
    }

    @Test
    public void size() throws Exception {
        assertThat(karutaIds.size(), is(3));
    }

    @Test
    public void contains() throws Exception {
        assertThat(karutaIds.contains(new KarutaIdentifier(1)), is(true));
    }

    @Test
    public void notContains() throws Exception {
        assertThat(karutaIds.contains(new KarutaIdentifier(4)), is(false));
    }
}

File: domain/src/test/java/me/rei_m/hyakuninisshu/domain/model/karuta/KarutaIdsTest.kt
/*
 * Copyright (c) 2018. Rei Matsushita
 *
 * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
 * compliance with the License. You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed under the License is
 * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
 * the License for the specific language governing permissions and limitations under the License.
 */

/* ktlint-disable package-name */
package me.rei_m.hyakuninisshu.domain.model.karuta

import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class KarutaIdsTest {

    private lateinit var karutaIds: KarutaIds

    @Before
    fun setUp() {
        karutaIds = KarutaIds(arrayListOf(
            KarutaIdentifier(1),
            KarutaIdentifier(2),
            KarutaIdentifier(3)
        ))
    }

    @Test
    fun createInstance() {
        val expected = arrayListOf(
            KarutaIdentifier(1),
            KarutaIdentifier(2),
            KarutaIdentifier(3)
        )
        assertThat(karutaIds.values).isEqualTo(expected)
        assertThat(karutaIds.size).isEqualTo(3)
    }

    @Test
    fun asRandomized() {
        assertThat(karutaIds.asRandomized).containsAll(karutaIds.values)
    }

    @Test
    fun contains() {
        assertThat(karutaIds.contains(KarutaIdentifier(1))).isTrue()
    }

    @Test
    fun notContains() {
        assertThat(karutaIds.contains(KarutaIdentifier(4))).isFalse()
    }
}
